### PR TITLE
fix: no data to plot is shown sometimes incorrectly [WEB-1413]

### PR DESCRIPTION
**Pull request by julian-determined-ai**
## Description

Currently, the metric charts in the `Compare` tab may flash no data to plot because the current check only ensures that the data is at least partially loaded. However, some metric charts might not have an updated data `series` causing the UI to show the `no data to plot` message. 

This PR includes an update to `useTrialMetrics` to return whether or not each individual metric contains data. The `Compare` metrics tab now uses that information to ensure that each metric's chart has been fully updated.  

## Test Plan

1. Open the `Compare` tab in the new experiments list. 
2. Select experiments that do not all contain the same metrics. For example, select an experiment that has only `validation loss` then select another experiment that has `validation accuracy`. Ensure that that as the new charts for the new metrics added, `No Data To Plot` is not flashed. You should see a loading spinner until the charts have rendered with data.
You can reference: https://github.com/determined-ai/release-party-issues/issues/119#issuecomment-1611880770 to see what the undesired behavior looks like.

3. De-select the experiments from (2) one by one. Again, ensure that any metric charts do no show "no data to plot" if data is available and instead always show a loading spinner.

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
WEB-1413

### CODE REVIEW

File: webui/react/src/pages/TrialDetails/useTrialMetrics.ts [hunk 1]
```diff
@@ -69,13 +70,17 @@ const summarizedMetricToSeries = (
...
-  );
-  return { data: trialData, hasData };
+  // Record whether or not each metric contains at least one value for any
+  // xAxis option.
+  Object.keys(trialData).forEach((key) => {
+    if (!metricHasData?.[key]) {
+      metricHasData[key] = xAxisOptions.some(
+        (xAxis) => (trialData?.[key]?.data?.[xAxis]?.length ?? 0) > 0,
+      );
+    }
```
Comment (ashtonG) [inline webui/react/src/pages/TrialDetails/useTrialMetrics.ts:81]: see above re: or operator:
```suggestion
    metricHasData[key] ||= xAxisOptions.some(
      (xAxis) => (trialData?.[key]?.data?.[xAxis]?.length ?? 0) > 0,
    );
```

File: webui/react/src/pages/TrialDetails/useTrialMetrics.ts [hunk 2]
```diff
@@ -141,18 +145,22 @@ export const useTrialMetrics = (
...
         response.forEach((r) => {
-          const { data: trialData, hasData } = summarizedMetricToSeries(r?.metrics, metrics);
-          if (hasData) {
-            anyTrialHasData = true;
-          }
+          const { data: trialData, metricHasData } = summarizedMetricToSeries(r?.metrics, metrics);
+          Object.keys(metricHasData).forEach((key) => {
+            if (!metricsHaveData?.[key]) {
+              metricsHaveData[key] = metricHasData[key];
+            }
```
Comment (ashtonG) [inline webui/react/src/pages/TrialDetails/useTrialMetrics.ts:152]:
```suggestion
            metricsHaveData[key] ||= metricHasData[key];
```

**Review by ashtonG — COMMENTED** [review]
looks fine functionally, had a style issue

**Comment by julian-determined-ai** [comment]
@ashtonG I implemented the styling updates.

**Review by ashtonG — APPROVED** [review]
